### PR TITLE
Add [junos_l2_interfaces] model

### PR DESCRIPTION
**Pull request by ganeshrn**

### CODE REVIEW

File: models/junos/l2_interfaces/junos_l2_interfaces.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+              - List of VLANs to be configured in trunk port. It's used as the VLAN range to ADD or
+                REMOVE from the trunk.
+              type: list
+            native_vlan:
+              description:
+              - Native VLAN to be configured in trunk port. It is used as the trunk native VLAN ID.
+              type: str
+        filter:
+          description:
+          - Packet filtering applicable for ethernet-switching
```
Comment (samccann) [inline models/junos/l2_interfaces/junos_l2_interfaces.yaml]:
```suggestion
          - Packet filtering applicable for ethernet-switching.
```

File: models/junos/l2_interfaces/junos_l2_interfaces.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+          suboptions:
+            input:
+              description:
+              - The name of input filter of ethernet-switching.
```
Comment (samccann) [inline models/junos/l2_interfaces/junos_l2_interfaces.yaml]:
```suggestion
              - The name of the input filter of ethernet-switching.
```

File: models/junos/l2_interfaces/junos_l2_interfaces.yaml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            output:
+              description:
+              - The name of output filter of ethernet-switching.
```
Comment (samccann) [inline models/junos/l2_interfaces/junos_l2_interfaces.yaml]:
```suggestion
              - The name of the output filter of ethernet-switching.
```

File: models/junos/l2_interfaces/deleted_example_01.txt [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+#}
+#ge-0/0/2 {
+#    description "non L2 interface";
+#    unit 0 {
+#        family inet {
+#            address 192.168.56.14/24;
+#        }
+#    }
+
+- name: Delete JUNOS L2 interfaces as in given arguments.
```
Comment (samccann) [inline models/junos/l2_interfaces/deleted_example_01.txt]:
```suggestion
- name: Delete L2 attributes of given interfaces (Note: This won't delete the interface itself).
```

Comment (danielmellado) [reply models/junos/l2_interfaces/deleted_example_01.txt]: Maybe we can just say that it deletes the logical interfaces (units)

File: models/junos/l2_interfaces/deleted_example_01.txt [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    state: deleted
+
+# After state:
+# ------------
+#
+# ansible@junos01# show interfaces
+# ge-0/0/1 {
+#    description "L2 interface";
+#    speed 1g;
+#
```
Comment (danielmellado) [inline models/junos/l2_interfaces/deleted_example_01.txt]: I think this is missing a closing key }

File: models/junos/l2_interfaces/junos_l2_interfaces.yaml [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    config:
+      description: A dictionary of Layer-2 interface options
+      type: list
+      elements: dict
+      suboptions:
+        name:
+          description:
+          - Full name of interface, e.g. ge-0/0/1.
+          type: str
+          required: True
```
Comment (danielmellado) [inline models/junos/l2_interfaces/junos_l2_interfaces.yaml]: Do you really think that this should be required? IMHO we can just make this optional.

Comment (danielmellado) [reply models/junos/l2_interfaces/junos_l2_interfaces.yaml]: nvm, I misread this as interface's description!

File: models/junos/l2_interfaces/overridden_example_01.txt [hunk 7]
```diff
@@ -0,0 +1,71 @@
+# Using replaced
```
Comment (danielmellado) [inline models/junos/l2_interfaces/overridden_example_01.txt]: this should be overridden

**Pull request merged**